### PR TITLE
DrWavAudioImporter: WAV loader plugin using dr_wav

### PR DESCRIPTION
**Pull request by Alicemargatroid**
New plugin providing extensive loading of WAV files including:

- [x] Channels beyond Stereo up to 7.1 surround sound
- [x] Better handling of errors and exceptions, allowing recovery in many more cases
- [x] Full support for 32 and 64 bit floating point audio
- [x] Support for all standard formats besides ADPCM

### CODE REVIEW

File: src/MagnumPlugins/DrWavAudioImporter/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+# Objects shared between plugin and test library
+add_library(DrWavAudioImporterObjects OBJECT
+    ${DrWavAudioImporter_SRCS}
+    ${DrWavAudioImporter_HEADERS})
+target_include_directories(DrWavAudioImporterObjects PUBLIC
+    $<TARGET_PROPERTY:Magnum::Audio,INTERFACE_INCLUDE_DIRECTORIES>
+    ${PROJECT_SOURCE_DIR}/src
+    ${PROJECT_BINARY_DIR}/src)
+target_compile_definitions(DrWavAudioImporterObjects PRIVATE "DrWavAudioImporterObjects_EXPORTS")
+# Include the stb_*.h files as a system directory to supress warnings
```
Comment (mosra) [inline src/MagnumPlugins/DrWavAudioImporter/CMakeLists.txt]: Very minor thing: this still references the `stb_*` files, should be `dr_*` I think ;)

File: src/MagnumPlugins/DrWavAudioImporter/Test/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    emscripten_embed_file(DrWavAudioImporterTest stereo16.wav "/stereo16.wav")
+    emscripten_embed_file(DrWavAudioImporterTest stereo24.wav "/stereo24.wav")
+
+    emscripten_embed_file(DrWavAudioImporterTest mono32f.wav "/mono32f.wav")
+    emscripten_embed_file(DrWavAudioImporterTest stereo32f.wav "/stereo32f.wav")
+
+    emscripten_embed_file(DrWavAudioImporterTest stereo64f.wav "/stereo64f.wav")
+
+    emscripten_embed_file(DrWavAudioImporterTest surround616.wav "/surround616.wav")
+endif()
```
Comment (mosra) [inline src/MagnumPlugins/DrWavAudioImporter/Test/CMakeLists.txt]: This list is missing some files and that's I guess why the CI Emscripten build fails.

File: CMakeLists.txt [hunk 3]
```diff
@@ -60,6 +60,7 @@ option(WITH_STBPNGIMAGECONVERTER "Build StbPngImageConverter plugin" OFF)
 option(WITH_STBTRUETYPEFONT "Build StbTrueTypeFont plugin" OFF)
 option(WITH_STBVORBISAUDIOIMPORTER "Build StbVorbisAudioImporter plugin" OFF)
 option(WITH_DRFLACAUDIOIMPORTER "Build DrFlacAudioImporter plugin" OFF)
+option(WITH_DRWAVAUDIOIMPORTER "Build DrWavAudioImporter plugin" ON)
```
Comment (Alicemargatroid) [inline CMakeLists.txt]: @mosra My bad, I had that on for testing! Will fix when I get home, apologies.

File: src/MagnumPlugins/DrWavAudioImporter/Test/CMakeLists.txt (hunk 2 above)
Comment (Alicemargatroid) [inline src/MagnumPlugins/DrWavAudioImporter/Test/CMakeLists.txt]: @mosra Always forget that. I'll patch that tonight

File: src/MagnumPlugins/DrWavAudioImporter/CMakeLists.txt (hunk 1 above)
Comment (Alicemargatroid) [inline src/MagnumPlugins/DrWavAudioImporter/CMakeLists.txt]: @mosra No problem, guess my replace all missed it

**Comment by Alicemargatroid** [comment]
@mosra Looks like the large size of one of the WAV files was the main cause for the failures (along with not specifying the files correctly). All fixed!

**Comment by mosra** [comment]
Ow, 8 MB? I missed that.

I'll squash the commits into one to avoid having that big file in Git history. Too bad the same file got already committed into the `magnum` repository :/

**Comment by mosra** [comment]
Cherry-picked in 4d9243fb91eadf8f19ea6e24140a212390dd64c6. I made some minor corrections and documentation updates in the following commits, you might want to check them because I think some of them will apply also to the FLAC importer, as a lot of code is common.

Thanks a lot for your contributions!

**Pull request closed without merging**